<?xml version="1.0"?>
<!-- This launch file uses the basic Grasp controller plugin. -->

<launch>

<include file="$(find vigir_robotiq_ros_control)/launch/l_robotiq_ros_controller.launch"/>
<!-- -->
<node ns="left_grasp_controller" name="vigir_grasp_controller" pkg="vigir_grasp_controllers_old" type="robotiq_grasp_controller_old_node"    output="screen" respawn="true">launch-prefix="xterm -e gdb -ex r"

    <param name="hand" value="l_hand" />

    <remap from="template_selection" to="/template/l_hand_template_match_request" />
    <remap from="hand_offset"        to="/template/l_hand_template_offset" />
    <remap from="mode_command"       to="/grasp_control/l_hand/mode_command" />
    <remap from="grasp_selection"    to="/grasp_control/l_hand/grasp_selection" />
    <remap from="release_grasp"      to="/grasp_control/l_hand/release_grasp" />

    <remap from="active_state"       to="/grasp_control/l_hand/active_state" />
    <remap from="grasp_status"       to="/grasp_control/l_hand/grasp_status" />
    <remap from="hand_mass"          to="/grasp_control/l_hand/hand_mass" />
    <remap from="robotiq_states"      to="/grasp_control/l_hand/tactile_feedback" />
    <remap from="wrist_target"       to="/grasp_control/l_hand/target_pose" />
    <remap from="template_stitch_pose"       to="/grasp_control/l_hand/template_stitch_pose" />
    
    <remap from="wrist_plan"         to="/grasp_control/l_hand/plan_request" />
    <remap from="template_stitch"    to="/grasp_control/l_hand/template_stitch_request" />
    <remap from="wrist_pose"         to="/flor/l_arm_current_pose" />
    <remap from="planner_status"     to="/grasp_planning/l_hand/status" />
    <remap from="controller_mode"    to="/flor/controller/mode" />
</node>
</launch>
